src/VisualStudio/Core/Def/Implementation/TaskList/CommentTaskTokenSerializer.cs

```diff
@@ -14,21 +16,29 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.TaskList
     [Export(typeof(IOptionPersister))]
     internal class CommentTaskTokenSerializer : IOptionPersister
     {
-        private readonly ITaskList _taskList;
+        private readonly Shell.IAsyncServiceProvider _serviceProvider;
         private readonly IOptionService _optionService;
 
+        private ITaskList _taskList;
         private string _lastCommentTokenCache = null;
 
         [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
         public CommentTaskTokenSerializer(
-            VisualStudioWorkspace workspace,
-            [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+            VisualStudioWorkspace workspace, [Import(typeof(SAsyncServiceProvider))] Shell.IAsyncServiceProvider serviceProvider)
         {
             _optionService = workspace.Services.GetService<IOptionService>();
+            _serviceProvider = serviceProvider;
+        }
+
+        public async System.Threading.Tasks.Task InitializeAsync(CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
 
             // The SVsTaskList may not be available or doesn't actually implement ITaskList
             // in the "devenv /build" scenario
-            _taskList = serviceProvider.GetService(typeof(SVsTaskList)) as ITaskList;
+            _taskList = (ITaskList)await _serviceProvider.GetServiceAsync(typeof(SVsTaskList)).ConfigureAwait(false);
```

Comment:
❗️ `ConfigureAwait(true)` is required.

❗️ Missing `JoinableTaskFactory.SwitchToMainThreadAsync()` before this line.